### DISCUSSION TITLE
Is there a way to do non-strict equality check between string and number (or general primitives)?

### DISCUSSION THREAD

**Discussion by damafeez** [vitest-dev/vitest#2503] [Q&A]
Is there something like [chai .equal](https://www.chaijs.com/api/assert/#method_equal) that lets you do `==` comparison?
Something like `expect('3').equal(3)`. This is particularly useful when reading text from the dom and comparing it to a value in memory which may be a number etc.

**Comment by Mayvis** [comment]
In my opinion, there is no `expect('3').equal(3)` comparision method.
But if you really want this pattern, maybe you can use this pattern to check by yourself.
Hope this can help.

```js
const compare = 3 == '3'
expect(compare).toBeTruthy()
```

**Reply by damafeez** [reply]
Thanks for the response. I just wanted to verify there wasn't a built-in method that I missed.

**Reply by sheremet-va** [reply]
> Thanks for the response. I just wanted to verify there wasn't a built-in method that I missed. 

You can just use chai assertions with vitest.
